lib/synapse/haproxy.rb

```diff
@@ -742,6 +747,17 @@ def generate_backend_stanza(watcher, config)
           b = "\tserver #{backend_name} #{backend['host']}:#{backend['port']}"
           b = "#{b} cookie #{backend_name}" unless config.include?('mode tcp')
           b = "#{b} #{watcher.haproxy['server_options']}" if watcher.haproxy['server_options']
+          if backend.has_key?('weight')
+            # Check if server_options/haproxy_server_options already contains weight, if so log a warning
+            if watcher.haproxy.fetch('server_options', '').include? 'weight'
+              log.warn "synapse: weight is defined by server_options and by nerve"
+            end
+            if backend['haproxy_server_options'] and backend['haproxy_server_options'].include? 'weight'
```

Comment:
unfortunately, due to how the parsing logic in ServiceWatcher works today, `haproxy_server_options` always exists as a key in backend and defaults to value `nil`